rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -41,8 +41,38 @@ static FORCE_INLINE bool cmp_rdata(const decltype(spu_thread::rdata)& lhs, const
 	return !(r._u64[0] | r._u64[1]);
 }
 
+//#if defined(_MSC_VER)
+//#define AVX_FUNC
+//#else
+//#define AVX_FUNC __attribute__((__target__("avx")))
+//#endif // _MSC_VER
+
+template <bool aligned_32 = false>
 static FORCE_INLINE void mov_rdata(decltype(spu_thread::rdata)& dst, const decltype(spu_thread::rdata)& src)
```

Comment:
First, template argument is completely useless, because there is no difference in performance between two variants. The only difference is how it reacts on unaligned access, which doesn't matter.
Second, to be compatible with AVX without AVX2, it should use `_mm256_storeu_ps`	and `_mm256_loadu_ps`.
Third, AVX path should be moved to a separate function in order to use AVX_FUNC attribute correctly.
Then it should work.

Reply:
Nvm, I rewrote it to use inline assembler since AVX_FUNC attribute can't inline. Also removed AVX check in case of compiling with AVX support.